examples/gif_paintable/gif_paintable/frame.rs

```diff
@@ -0,0 +1,53 @@
+use std::time::Duration;
+
+use gtk::prelude::*;
+use gtk::{gdk, glib};
+
+pub struct Frame {
+    pub width: i32,
+    pub height: i32,
+    pub texture: gdk::Paintable,
+    pub frame_duration: Duration,
+}
+
+impl Frame {
+    pub fn new(f: image::Frame) -> Self {
+        let mut frame_duration = Duration::from(f.delay());
+
+        // convention is to use 100 milliseconds duration if it is defined as 0.
+        if frame_duration.is_zero() {
+            frame_duration = Duration::from_millis(100);
+        }
+
+        let image = f.into_buffer();
+
+        let samples = image.into_flat_samples();
+        let (stride, width, height) = samples.extents();
+
+        // glib::Bytes must contain gtk_stride * height number of bytes.
+        // since each pixel in the gif frame contains `stride` bytes
+        // of information we must map this stride to a value that gdk
+        // understands (see https://gtk-rs.org/gtk4-rs/git/docs/gdk4/struct.MemoryTexture.html)
+        let gtk_stride = stride * width;
```

Comment:
This would be `samples.layout.width_stride` FWIW. It's not guaranteed to be pixel/sample stride * width.

Reply:
Don't we just want to use `samples.layout.height_stride`? which would be the stride to the next row, aka rowstride?

Reply:
Erm yes. `height_stride` is what I meant, not `width_stride`.